src/librustc/query/mod.rs

```diff
@@ -461,6 +461,16 @@ rustc_queries! {
     }
 
     Other {
+        /// Obtain all the calls into other local functions
+        query local_calls(key: DefId) -> &'tcx [DefId] {
+            fatal_cycle
+            eval_always
```

Comment:
You probably want to cache this and not use `eval_always` so it doesn't cause `mir_validated` to be recomputed.